### PR TITLE
Add option to use a different Connector.

### PR DESCRIPTION
**Pull request by tgxworld**

### CODE REVIEW

**Comment by badboy** [comment]
What would be the use case of this?

**Comment by tgxworld** [comment]
I'm trying to run a custom connector without having to monkey patch. Basically have the ability to subclass `Redis::Client::Connector` easily.

**Comment by tgxworld** [comment]
@badboy I would like to clarify if the request is reasonable :)

**Comment by SamSaffron** [comment]
@badboy any news on this? need to know if its acceptable or not, I hate carrying freedom patches around for Discourse :)

**Comment by SamSaffron** [comment]
see: 

https://github.com/discourse/discourse/pull/4056

**Comment by SamSaffron** [comment]
@badboy any news on this issue?

**Comment by tgxworld** [comment]
ping @djanowski Any thoughts about this?

**Comment by tgxworld** [comment]
@badboy  @djanowski Don't mean to ping multiple times but we would love to get some feedback on this.

**Comment by tgxworld** [comment]
@byroot Was wondering if you might have any thoughts on this.

File: lib/redis/client.rb [hunk 1]
```diff
@@ -79,11 +79,14 @@ def initialize(options = {})
...
 
-      if options.include?(:sentinels)
-        @connector = Connector::Sentinel.new(@options)
-      else
-        @connector = Connector.new(@options)
-      end
+      @connector =
+        if options.include?(:sentinels)
+          Connector::Sentinel.new(@options)
+        elsif options.include?(:connector) && options[:connector].is_a?(Class)
```
Comment (byroot) [inline lib/redis/client.rb]: I wouldn't check for `is_a?(Class)`, `respond_to?(:new)` would be better as it would allow for facades etc.

**Comment by tgxworld** [comment]
@byroot Thank you for reviewing. I've updated the PR as per your comments and added tests as well.

**Comment by tgxworld** [comment]
@byroot Thank you for reviewing. Just curious, is Shopify/Rails taking up ownership of this gem now?

**Comment by byroot** [comment]
A few people from Shopify have been added as collaborators on the repo indeed. We should also get Rubygems access some day so we can release new versions as well.

**Comment by tgxworld** [comment]
Awesome :+1: Thanks for reviewing once again!

**Pull request merged**